### DISCUSSION TITLE
Cannot customize Hireable in a custom locationtype with a custom faction

### DISCUSSION THREAD

**Discussion by NotAlwaysTrue** [FakeFishGames/Barotrauma#14570] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Here is a F3 console snapshot
![6(5IPC7IWVU$B}~VIDH~ZK3](https://github.com/user-attachments/assets/f4543013-a82c-46b1-92d4-b79601849d3c)

The code is as follows
```
  <tsm_city
    identifier="tsm_city"
    replaceinradiation="Abandoned"
    nameidentifiers="city"
    commonnessperzone="1: 50, 2: 50, 3: 50, 4: 50, 5:50"
    mincountperzone="1: 1"
    faction="tsm"
    AllowAsBiomeGate="true"><!--0.4 0.5 0.7 0.2 0.3-->

    <Symbol texture="%ModDir%/faction/tsm.png" sourcerect="128,16,103,112" color="255,255,255,255"/>
    <RadiationSymbol texture="Content/Map/MapAtlas.png" sourcerect="896,512,128,128" color="245,105,105"/>
    <Portrait texture="%ModDir%/Map/冰面.png" sourcerect="0,0,0,0" premultiplyalpha="false"/>

    <Hireable identifier="engineer" commonness="5" AlwaysAvailableIfMissingFromCrew="true"/>
    <Hireable identifier="mechanic" commonness="5" AlwaysAvailableIfMissingFromCrew="true"/>
    <Hireable identifier="assistant" commonness="8" AlwaysAvailableIfMissingFromCrew="true"/>
    <Hireable identifier="medicaldoctor" commonness="5" AlwaysAvailableIfMissingFromCrew="true"/>    
    <Hireable identifier="securityofficer" commonness="5" AlwaysAvailableIfMissingFromCrew="true"/>
    <Hireable identifier="captain" commonness="5" AlwaysAvailableIfMissingFromCrew="true"/>

    <Store
      maxreputationmodifier="0.15"
      sellpricemodifier="0.3"
      dailyspecialpricemodifier="0.7"
      requestgoodpricemodifier="2"
      initialbalance="10000"
      pricemodifierrange="5"
      dailyspecialscount="5"
      requestedgoodscount="3" />
  </tsm_city>

``` 

However you can see nothing related to NPCSets was defined in SharedSource/Map/Map/LocationType.cs, case "hireable", there seemingly have no way to tell hiremanager which NPCSet to choose from. 

Please correct me if anything I did is wrong

### Reproduction steps

1. Create a custom faction
2. Create a custom locationtype with your custom faction
3. Copy paste and modify a random outpost generation parameter to a proper testing state
4. Start campion
5. Jump to this custom location
7. Open F3 and see these errors

### Bug prevalence

Everytime I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
Could not find human prefab "hireabletsmsecurityofficer" from "outpostnpcs1".
Couldn't create a hireable for the location: character prefab "hireabletsmsecurityofficer" not found in the NPC set "outpostnpcs1".
Could not find human prefab "hireabletsmengineer" from "outpostnpcs1".
Couldn't create a hireable for the location: character prefab "hireabletsmengineer" not found in the NPC set "outpostnpcs1".
Could not find human prefab "hireabletsmmechanic" from "outpostnpcs1".
Couldn't create a hireable for the location: character prefab "hireabletsmmechanic" not found in the NPC set "outpostnpcs1".
Could not find human prefab "hireabletsmmedic" from "outpostnpcs1".
Couldn't create a hireable for the location: character prefab "hireabletsmmedic" not found in the NPC set "outpostnpcs1".
Could not find human prefab "hireabletsmsecurityofficerveteran" from "outpostnpcs1".
Couldn't create a hireable for the location: character prefab "hireabletsmsecurityofficerveteran" not found in the NPC set "outpostnpcs1".
Could not find human prefab "hireabletsmengineerveteran" from "outpostnpcs1".
Couldn't create a hireable for the location: character prefab "hireabletsmengineerveteran" not found in the NPC set "outpostnpcs1".
Could not find human prefab "hireabletsmmechanicveteran" from "outpostnpcs1".
Couldn't create a hireable for the location: character prefab "hireabletsmmechanicveteran" not found in the NPC set "outpostnpcs1".
Could not find human prefab "hireabletsmmedicveteran" from "outpostnpcs1".
Couldn't create a hireable for the location: character prefab "hireabletsmmedicveteran" not found in the NPC set "outpostnpcs1".
```

**Comment by ubertpendragon** [comment]
It's somewhat easy with a custom faction as you just define the hireables there

![image](https://github.com/user-attachments/assets/d9d70fe7-1dcd-411b-8efe-f2401c7752dd)


The problem is two fold:

You can't define what NPCs can be hired at *specific* location types. 
Say I have a special military outpost and I want *specific* custom soldier NPCs to only be recruiteable there; not possible. 
You basically need to create a custom job, create your npcs with that job as hireable and add that hireable to the list

You need to define at least 1 "vanilla" hireable, so your hireable list can never be truly customized like
`<Hireable identifier="assistant" commonness="8" AlwaysAvailableIfMissingFromCrew="true"/>`
#13864

**Reply by NotAlwaysTrue** [reply]
I think you are right. I forgot something in my faction.Thanks for offering help
